README.rst
Python Client for reCAPTCHA Enterprise API
==========================================

|stable| |pypi| |versions|

`reCAPTCHA Enterprise API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-recaptcha-enterprise.svg
   :target: https://pypi.org/project/google-cloud-recaptcha-enterprise/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-recaptcha-enterprise.svg
   :target: https://pypi.org/project/google-cloud-recaptcha-enterprise/
.. _reCAPTCHA Enterprise API: https://cloud.google.com/recaptcha-enterprise
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/recaptchaenterprise/latest
.. _Product Documentation:  https://cloud.google.com/recaptcha-enterprise

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the reCAPTCHA Enterprise API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the reCAPTCHA Enterprise API.:  https://cloud.google.com/recaptcha-enterprise
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-recaptcha-enterprise


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-recaptcha-enterprise

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for reCAPTCHA Enterprise API
   to see other available methods on the client.
-  Read the `reCAPTCHA Enterprise API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _reCAPTCHA Enterprise API Product documentation:  https://cloud.google.com/recaptcha-enterprise
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

## [1.5.0](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v1.4.1...v1.5.0) (2022-01-10)


### Features

* add new reCAPTCHA Enterprise fraud annotations ([#163](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/163)) ([3c638f9](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/3c638f97a966efdc6809016b9a50d63ffe4b380c))


### Documentation

* **samples:** added samples and test to migrate key and get metrics ([#161](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/161)) ([4ab3ab8](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/4ab3ab8979d58e9b490792cdb966147fa97f28b6))

### [1.4.1](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v1.4.0...v1.4.1) (2021-11-16)


### Documentation

* **samples:** added sample and tests for annotate assessment API ([#155](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/155)) ([353f0ff](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/353f0ffa27119bf3a8a8f6cddecb41d1ca8dbd31))

## [1.4.0](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v1.3.1...v1.4.0) (2021-11-03)


### Features

* add reCAPTCHA Enterprise account defender API methods ([#146](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/146)) ([8149df9](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/8149df9e3038bf02483fb2ee0c9dfc9d713a6152))


### Documentation

* **samples:** removed assessment name in create_assessment sample ([#147](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/147)) ([f11134d](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/f11134da5e9e9770b73dadd96c08ba69c99b968f))

### [1.3.1](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v1.3.0...v1.3.1) (2021-11-01)


### Bug Fixes

* **deps:** drop packaging dependency ([22ed89d](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/22ed89dcaa628790d09584b3bd20c35115647bb7))
* **deps:** require google-api-core >= 1.28.0 ([22ed89d](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/22ed89dcaa628790d09584b3bd20c35115647bb7))


### Documentation

* list oneofs in docstring ([22ed89d](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/22ed89dcaa628790d09584b3bd20c35115647bb7))

## [1.3.0](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v1.2.0...v1.3.0) (2021-10-14)


### Features

* add support for python 3.10 ([#137](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/137)) ([fda16a4](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/fda16a44a84b173a8866090021e03eef0cb82025))


### Documentation

* **samples:** add reCAPTCHA Enterprise code samples  ([#112](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/112)) ([879acf1](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/879acf12d24b5148e372f0f76a243ea0fc66286e))

## [1.2.0](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v1.1.2...v1.2.0) (2021-10-08)


### Features

* add context manager support in client ([#131](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/131)) ([69d2b34](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/69d2b347f9aab3f30a3b0265fc9dcd9a3926f62d))

### [1.1.2](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v1.1.1...v1.1.2) (2021-09-30)


### Bug Fixes

* improper types in pagers generation ([cab9a71](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/cab9a71083b2482396ecd53051bb694937c7fe7d))

### [1.1.1](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v1.1.0...v1.1.1) (2021-09-24)


### Bug Fixes

* add 'dict' annotation type to 'request' ([7d49e68](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/7d49e6830e0f914d8fb2f20961a3ae6953244ed4))

## [1.1.0](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v1.0.0...v1.1.0) (2021-09-16)


### Features

* add GetMetrics and MigrateKey methods to reCAPTCHA enterprise API ([#119](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/119)) ([6c8bf2f](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/6c8bf2f7f9fabb7fb23257fe5978ba59160d6875))

## [1.0.0](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v0.4.2...v1.0.0) (2021-08-03)


### Features

* bump release level to production/stable ([#105](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/105)) ([fab3e48](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/fab3e48b6ddc31bed65e9bedfd9f06aa33fd2c02))

### [0.4.2](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v0.4.1...v0.4.2) (2021-07-28)


### Bug Fixes

* enable self signed jwt for grpc ([#101](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/101)) ([2a9cba8](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/2a9cba89af89d76a1e9a4922e2901bd4847de949))


### Documentation

* add Samples section to CONTRIBUTING.rst ([#97](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/97)) ([ffcc165](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/ffcc1651929337be437250e440b75548c453ced9))


### Miscellaneous Chores

* release as 0.4.2 ([#102](https://www.github.com/googleapis/python-recaptcha-enterprise/issues/102)) ([ff137b0](https://www.github.com/googleapis/python-recaptcha-enterprise/commit/ff137b00f5282dc061941b6645143db7c66a6718))

### [0.4.1](https://www.github.com/googleapis/python-recaptcha-enterprise/compare/v0.4.0...v0.4.1) (2021-07-20)


### Bug Fixes


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9', '3.10'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/recaptchaenterprise_v1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.9')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.9')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
